src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1150,6 +1152,58 @@ void Instance::HandleQueryIdentity(HandlerContext & ctx, const Commands::QueryId
 }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
 
+Instance * Instance::sInstanceRecord = nullptr;
+
+void Instance::RecordNetwork(Instance * currentNetwork)
+{
+    Instance * curr = sInstanceRecord;
+    while (curr)
+    {
+        if (curr == currentNetwork)
+        {
+            return;
+        }
+        curr = curr->mNext;
+    }
+
+    currentNetwork->mNext = sInstanceRecord;
+    sInstanceRecord       = currentNetwork;
+}
+
+void Instance::DisconnectOtherNetworks(Instance * currentNetwork)
+{
+    Instance * curr = sInstanceRecord;
+    while (curr)
+    {
+        if (curr != currentNetwork && curr->mpWirelessDriver)
+        {
+            uint8_t networkNo         = 0;
+            uint8_t maxNetworks       = curr->mpBaseDriver->GetMaxNetworks();
+            bool haveConnectedNetwork = false;
+            EnumerateAndRelease(curr->mpBaseDriver->GetNetworks(), [&](const Network & network) {
+                if (network.connected)
+                {
+                    haveConnectedNetwork = true;
+                    return Loop::Break;
+                }
+                ++networkNo;
+                return (networkNo == maxNetworks) ? Loop::Break : Loop::Continue;
+            });
+
+            if (!haveConnectedNetwork)
+            {
+                /* It's different to check the media driver connection state here.
+                 * Check the `Connected` flag in `NetworkInfoStruct` in stead, if not set,
+                 * exec `DisconnectFromNetwork` for driver. However, the physical media connection
+                 * may not necessarily exist in this case, such as after initialization.
+                 */
+                curr->mpWirelessDriver->DisconnectFromNetwork();
```

Comment:
Error handling?

Reply:
`DisconnectFromNetwork` failed may not affect subsequent connections to other networks. So I'm not sure if we should handle errors here. Use `ReturnOnFailure` temporarily.